oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/gateway/IGatewayService.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.receiver.trace.gateway;
+
+import org.apache.skywalking.oap.server.library.module.Service;
+
+/**
+ * @author kezhenxu94
+ */
+public interface IGatewayService extends Service {
```

Comment:
Why this is a service? Who would use it? Service is for inter-module.

Reply:
You could create a class to do all these things, but it is not a module service, actually. receiver usually doesn't include service definition. `ISegmentParserService` service exists because the zipkin receiver is going to use it.